spec/rspec/matchers/built_in/respond_to_spec.rb

```diff
@@ -493,6 +515,16 @@ def obj.some_method(arg, arg2, arg3, *args); end
       expect(obj).to respond_to(:foo).with_keywords(:a, :b)
     end
 
+    it 'passes if target is :new with keywords' do
+      # note we can't use the metaobject definition for initialize
+      klass = eval %{Class.new { def initialize(a: nil, b: nil); end}}
+      expect(klass).to respond_to(:new).with_keywords(:a, :b)
+
+      # note we can't use the metaobject definition for initialize
+      klass_2 = eval %{Class.new { def initialize(**kw_args); end}}
+      expect(klass_2).to respond_to(:new).with_keywords(:a, :b)
```

Comment:
Seems like we need some examples of the expectations failing. All the ones you've added to these specs are of passing expectations.

Reply:
Theres one failing, but you're right it might not be enough

Reply:
Something like?
```ruby
klass_3 = eval %{Class.new { def initialize(a: nil, b: nil); end}}
expect {
  expect(klass_3).to respond_to(:new).with_keywords(1..2)
}.to fail_with(/expected #<Class.*> to respond to :new with keyword 1..2/)
```